### PR TITLE
task 4 initial push

### PR DESCRIPTION
**Pull request by priyankaapatelgit**

### CODE REVIEW

File: app/src/App.js [hunk 1]
```diff
@@ -1,11 +1,21 @@
...
+
+
+
 
 class App extends Component {
+  state = {
+    blk: 'app',
+    elt: 'sub',
+    mods: ['class1', 'class2']
+  };
```
Comment (meghakarkihalli) [inline app/src/App.js]: why is this needed? 
Why can't we send the object directly to the function?

File: app/src/App.js [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
   render() {
     return (
       <div>
-        Welcome to Credit Tracking
+
+        <ClassNames {...this.state} />
```
Comment (meghakarkihalli) [inline app/src/App.js]: this should just be a utility function not a component

File: app/src/component-library/helpers/classNames/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        }
+
+        if (mods.length > 0) {
+            mods.map((item) => {
+                if (item !== '')
+                    subClass += ` ${blk}__${elt}--${item}`
+            })
+        }
+        return subClass;
+    }
```
Comment (meghakarkihalli) [inline app/src/component-library/helpers/classNames/index.js:20]: what happens if the input is `{ blk: 'abc' }`

**Comment by meghakarkihalli** [comment]
@Priyankaa-patel please address the comments.

**Review by meghakarkihalli — CHANGES_REQUESTED** [review]

File: app/src/App.js (hunk 1 above)
Comment (priyankaapatelgit) [reply app/src/App.js]: I created a component that's why I used state.  I will convert it in utility function and pass object in it.

Comment (meghakarkihalli) [reply app/src/App.js]: className({ blk: 'app', elt: 'sub', mods: [ 'class1', 'class2' ]})

File: app/src/component-library/helpers/classNames/index.js (hunk 3 above)
Comment (priyankaapatelgit) [reply app/src/component-library/helpers/classNames/index.js:20]: The output is: abc

File: app/src/component-library/helpers/classNames/index.js [hunk 4]
```diff
@@ -1,14 +1,17 @@
-export const ClassNames = ({ blk = '', elt = '', mods = [''] }) => {
+export const classNames = ({ blk = '', elt = '', mods = [''] }) => {
```
Comment (meghakarkihalli) [inline app/src/component-library/helpers/classNames/index.js:1]: mods = []

**Review by meghakarkihalli — APPROVED** [review]
